pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -306,15 +307,66 @@ func (r *NSTemplateSetReconciler) ensureInnerNamespaceResources(logger logr.Logg
 		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err, "failed to update namespace '%s'", nsName)
 	}
 
-	log.Info("namespace provisioned with required resources", "namespace", tcNamespace)
+	log.Info("namespace provisioned with required resources", "tier", nsTmplSet.Spec.TierName, "namespace", tcNamespace)
 
 	// TODO add validation for other objects
 	return nil
 }
 
+func (r *NSTemplateSetReconciler) ensureClusterResources(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet) error {
+	logger.Info("creating cluster resources", "username", nsTmplSet.GetName(), "tier", nsTmplSet.Spec.TierName)
+	clusterResources := nsTmplSet.Spec.ClusterResources
+	// cluster-wide resources are optional. Skip if none was specified in the given NSTemplateSet
+	if clusterResources == nil {
+		return nil
+	}
+
+	tmpl, err := r.getTemplateContent(nsTmplSet.Spec.TierName, ClusterResources)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusClusteResourcesProvisionFailed, err, "failed to to retrieve template for the cluster resources")
+	}
+
+	tmplProcessor := template.NewProcessor(r.client, r.scheme)
+	params := map[string]string{"USERNAME": nsTmplSet.GetName()}
+	objs, err := tmplProcessor.Process(tmpl, params)
+	logger.Info("cluster resources template", "objects", objs)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusClusteResourcesProvisionFailed, err, "failed to to retrieve template for the cluster resources")
+	}
+
+	for _, rawObj := range objs {
+		acc, err := meta.Accessor(rawObj.Object)
+		if err != nil {
+			return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusClusteResourcesProvisionFailed, err, "invalid element in template for the cluster resources")
+		}
+
+		// set labels
+		labels := acc.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels[toolchainv1alpha1.OwnerLabelKey] = nsTmplSet.GetName()
+		labels[toolchainv1alpha1.TypeLabelKey] = ClusterResources
+		labels[toolchainv1alpha1.TierLabelKey] = nsTmplSet.Spec.TierName
+		acc.SetLabels(labels)
+
+		// Note: we don't see an owner reference between the NSTemplateSet (namespaced resource) and the cluster-wide resources
+		// because a namespaced resource (NSTemplateSet) cannot be the owner of a cluster resource (the GC will delete the child resource, considering it is an orphan resource)
+		// As a consequence, when the NSTemplateSet is deleted, we explicitly delete the associated namespaces that belong to the same user.
+		// see https://issues.redhat.com/browse/CRT-429
+
```

Comment:
I didn't check so ignore this comment if you have already added such a test but let's make sure we do have an e2e test which makes sure we delete the corresponding cluster resources when the corresponding user account is deleted.

Reply:
thanks for pointing this! done in 4c7fe97

Reply:
Thanks! What about adding an e2e test for that?

Reply:
yes, it's planned, too ;) (will work on it today, actually)